diff --git a/power_monitor/Android.bp b/power_monitor/Android.bp
index 524ba3aaa..dfb12af3e 100644
--- a/power_monitor/Android.bp
+++ b/power_monitor/Android.bp
@@ -12,6 +12,13 @@ package {
     default_applicable_licenses: ["external_crosvm_license"],
 }

+genrule {
+    name: "copy_power_monitor_build_out",
+    srcs: ["out/*"],
+    cmd: "cp $(in) $(genDir)",
+    out: ["generated.rs"],
+}
+
 rust_library {
     name: "libpower_monitor",
     defaults: ["crosvm_inner_defaults"],
@@ -20,6 +27,7 @@ rust_library {
     cargo_env_compat: true,
     cargo_pkg_version: "0.1.0",
     crate_root: "src/lib.rs",
+    srcs: [":copy_power_monitor_build_out"],
     edition: "2021",
     rustlibs: [
         "libbase_rust",
