diff --git a/Android.bp b/Android.bp
index 509b3b73d..163a24305 100644
--- a/Android.bp
+++ b/Android.bp
@@ -37,6 +37,7 @@ rust_binary {
     name: "crosvm",
     defaults: ["crosvm_inner_defaults"],
     host_supported: true,
+    prefer_rlib: true,
     crate_name: "crosvm",
     cargo_env_compat: true,
     cargo_pkg_version: "0.1.0",
@@ -54,7 +55,6 @@ rust_binary {
         "gdbstub_arch",
         "geniezone",
         "gfxstream",
-        "gfxstream_stub",
         "gpu",
         "gpu_display",
         "gunyah",
@@ -135,6 +135,8 @@ rust_binary {
         },
         android: {
             shared_libs: [
+                "libbinder_ndk",
+                "libnativewindow",
                 "libprocessgroup",
             ],
         },
@@ -143,6 +145,18 @@ rust_binary {
                 "libprocessgroup",
             ],
         },
+        linux_bionic_arm64: {
+            relative_install_path: "aarch64-linux-bionic",
+        },
+        linux_glibc_x86_64: {
+            relative_install_path: "x86_64-linux-gnu",
+        },
+        linux_musl_x86_64: {
+            relative_install_path: "x86_64-linux-musl",
+        },
+        linux_musl_arm64: {
+            relative_install_path: "aarch64-linux-musl",
+        },
     },
     ld_flags: [
         "-Wl,--rpath,\\$$ORIGIN",
@@ -177,7 +191,6 @@ rust_test {
         "gdbstub_arch",
         "geniezone",
         "gfxstream",
-        "gfxstream_stub",
         "gpu",
         "gpu_display",
         "gunyah",
