diff --git a/disk/Android.bp b/disk/Android.bp
index ef8395e82..088ac60b8 100644
--- a/disk/Android.bp
+++ b/disk/Android.bp
@@ -28,19 +28,24 @@ rust_test {
     edition: "2021",
     features: [
         "android-sparse",
+        "composite-disk", // Added manually
         "qcow",
     ],
     rustlibs: [
         "libbase_rust",
         "libcfg_if",
+        "libcrc32fast", // Added manually
         "libcros_async",
         "libdata_model",
         "libfutures",
         "liblibc",
+        "libprotobuf", // Added manually
+        "libprotos", // Added manually
         "libserde",
         "libsync_rust",
         "libtempfile",
         "libthiserror",
+        "libuuid", // Added manually
         "libvm_memory",
         "libzerocopy",
     ],
@@ -61,18 +66,23 @@ rust_library {
     edition: "2021",
     features: [
         "android-sparse",
+        "composite-disk", // Added manually
         "qcow",
     ],
     rustlibs: [
         "libbase_rust",
         "libcfg_if",
+        "libcrc32fast", // Added manually
         "libcros_async",
         "libdata_model",
         "libfutures",
         "liblibc",
+        "libprotobuf", // Added manually
+        "libprotos", // Added manually
         "libserde",
         "libsync_rust",
         "libthiserror",
+        "libuuid", // Added manually
         "libvm_memory",
         "libzerocopy",
     ],
