diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index f2c450cb4..a32b9445d 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -29,12 +29,26 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -61,10 +75,24 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-        "libgbm",
-        "libvirglrenderer",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
