diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index c69e8a386..39f7306a6 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -23,7 +23,6 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -35,14 +34,73 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    shared_libs: [
+        "libc++",
+        "libbase",
+    ],
+    target: {
+        android: {
+            shared_libs: [
+                "libnativewindow",
+                "libcutils",
+                "liblog",
+            ],
+        },
+        host: {
+            features: ["vulkano"],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
     static_libs: [
+        "libdrm",
+        "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
     ],
+}
+
+rust_library {
+    name: "librutabaga_gfx_gfxstream",
+    defaults: ["crosvm_inner_defaults"],
+    host_supported: true,
+    vendor_available: true,
+    crate_name: "rutabaga_gfx",
+    cargo_env_compat: true,
+    cargo_pkg_version: "0.1.2",
+    srcs: ["src/lib.rs"],
+    edition: "2021",
+    rustlibs: [
+        "libcfg_if",
+        "liblibc",
+        "liblog_rust",
+        "libnix",
+        "libthiserror",
+        "libzerocopy",
+    ],
+    proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    features: [
+        "gfxstream",
+    ],
     shared_libs: [
-        "libdrm",
-        "libepoxy",
+        "libgfxstream_backend",
     ],
+    target: {
+        host: {
+            features: ["vulkano"],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -61,7 +119,6 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -73,12 +130,13 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    static_libs: [
+    shared_libs: [
+        "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
     ],
-    shared_libs: [
+    static_libs: [
         "libdrm",
-        "libepoxy",
     ],
 }
