diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index 3384aee72..55483d542 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -22,7 +22,6 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -34,14 +33,46 @@ rust_library {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
     shared_libs: [
         "libdrm",
         "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
     ],
 }
 
+rust_library {
+    name: "librutabaga_gfx_gfxstream",
+    defaults: ["crosvm_inner_defaults"],
+    host_supported: true,
+    vendor_available: true,
+    crate_name: "rutabaga_gfx",
+    cargo_env_compat: true,
+    cargo_pkg_version: "0.1.2",
+    srcs: ["src/lib.rs"],
+    edition: "2021",
+    rustlibs: [
+        "libcfg_if",
+        "liblibc",
+        "liblog_rust",
+        "libnix",
+        "libthiserror",
+        "libzerocopy",
+    ],
+    proc_macros: ["libremain"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    features: ["gfxstream"],
+    shared_libs: [
+        "libgfxstream_backend",
+    ],
+}
+
 rust_test {
     name: "rutabaga_gfx_test_src_lib",
     defaults: ["crosvm_inner_defaults"],
@@ -58,7 +89,6 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -74,6 +104,7 @@ rust_test {
         "libdrm",
         "libepoxy",
         "libgbm",
+        "libgfxstream_backend",
         "libvirglrenderer",
     ],
 }
