diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index 8c19dbfff..8220b1c5d 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -23,7 +23,6 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -38,13 +37,80 @@ rust_library {
     ],
     proc_macros: ["libremain"],
     static_libs: [
-        "libgbm",
+        "libepoxy",
+        "libgfxstream_backend",
         "libvirglrenderer",
     ],
     shared_libs: [
-        "libdrm",
-        "libepoxy",
+        "libc++",
+        "libbase",
+    ],
+    apex_available: ["com.android.virt"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    target: {
+        android: {
+            shared_libs: [
+                "libnativewindow",
+                "libcutils",
+                "liblog",
+            ],
+        },
+        host: {
+            features: ["vulkano"],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
+}
+
+rust_library {
+    name: "librutabaga_gfx_gfxstream",
+    host_supported: true,
+    vendor_available: true,
+    crate_name: "rutabaga_gfx",
+    cargo_env_compat: true,
+    cargo_pkg_version: "0.1.2",
+    srcs: ["src/lib.rs"],
+    edition: "2021",
+    rustlibs: [
+        "libcfg_if",
+        "liblibc",
+        "liblog_rust",
+        "libnix",
+        "libserde",
+        "libserde_json",
+        "libthiserror",
+        "libzerocopy",
     ],
+    proc_macros: ["libremain"],
+    target: {
+        host: {
+            cfgs: [
+                "fence_passing_option1",
+                "gfxstream_unstable",
+            ],
+            features: [
+                "gfxstream",
+                "vulkano",
+            ],
+            shared_libs: [
+                "libgfxstream_backend",
+            ],
+            compile_multilib: "64",
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+        android: {
+            cfgs: [
+                "fence_passing_option1",
+                "gfxstream_unstable",
+            ],
+        },
+    },
     apex_available: ["com.android.virt"],
 }
 
@@ -64,7 +130,6 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
         "virgl_renderer",
     ],
     rustlibs: [
@@ -79,12 +144,9 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
-    static_libs: [
-        "libgbm",
-        "libvirglrenderer",
-    ],
     shared_libs: [
-        "libdrm",
         "libepoxy",
+        "libgfxstream_backend",
+        "libvirglrenderer",
     ],
 }
