diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index 05d589ae..77c154be 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -33,14 +33,26 @@ rust_library {
         "libthiserror",
     ],
     proc_macros: ["libremain"],
-    static_libs: [
-        "libgbm",
-        "libvirglrenderer",
-    ],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
     apex_available: [
         "//apex_available:platform",
         "//apex_available:anyapex",
@@ -75,12 +86,24 @@ rust_test {
         "libthiserror",
     ],
     proc_macros: ["libremain"],
-    static_libs: [
-        "libgbm",
-        "libvirglrenderer",
-    ],
-    shared_libs: [
-        "libdrm",
-        "libepoxy",
-    ],
+    target: {
+        host: {
+            features: ["gfxstream"],
+            shared_libs: [
+                "libgfxstream_backend",
+                "libvirglrenderer",
+            ],
+        },
+        android: {
+            shared_libs: [
+                "libcutils",
+                "libdrm",
+            ],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
 }
