diff --git a/riscv64/Android.bp b/riscv64/Android.bp
index 7d065f875..9e14e8616 100644
--- a/riscv64/Android.bp
+++ b/riscv64/Android.bp
@@ -19,6 +19,28 @@ rust_library {
         "gdbstub",
         "gdbstub_arch",
     ],
+    rustlibs: [
+        "libarch",
+        "libbase_rust",
+        "libcros_fdt",
+        "libdata_model",
+        "libdevices",
+        "libgdbstub",
+        "libgdbstub_arch",
+        "libhypervisor",
+        "libkernel_cmdline",
+        "libkvm",
+        "libkvm_sys",
+        "liblibc",
+        "libminijail_rust",
+        "librand",
+        "libresources",
+        "libsync_rust",
+        "libthiserror",
+        "libvm_control",
+        "libvm_memory",
+    ],
+    proc_macros: ["libremain"],
 }
 
 rust_test {
@@ -40,4 +62,26 @@ rust_test {
         "gdbstub",
         "gdbstub_arch",
     ],
+    rustlibs: [
+        "libarch",
+        "libbase_rust",
+        "libcros_fdt",
+        "libdata_model",
+        "libdevices",
+        "libgdbstub",
+        "libgdbstub_arch",
+        "libhypervisor",
+        "libkernel_cmdline",
+        "libkvm",
+        "libkvm_sys",
+        "liblibc",
+        "libminijail_rust",
+        "librand",
+        "libresources",
+        "libsync_rust",
+        "libthiserror",
+        "libvm_control",
+        "libvm_memory",
+    ],
+    proc_macros: ["libremain"],
 }
