diff --git b/gpu_display/Android.bp a/gpu_display/Android.bp
index 989d1acd..cd5038f4 100644
--- b/gpu_display/Android.bp
+++ a/gpu_display/Android.bp
@@ -34,7 +34,19 @@ rust_library {
     ],
     proc_macros: ["libremain"],
     static_libs: ["libdisplay_wl"],
-    shared_libs: ["libwayland-client"],
+
+    // added manually
+    target: {
+        host: {
+            shared_libs: ["libwayland_client"],
+        },
+        android: {
+            static_libs: [
+                "libwayland_client_static",
+                "libffi",
+            ],
+        },
+    },
 }
 
 cc_library_static {
